### PR TITLE
faad2: update to 2.10.0

### PR DESCRIPTION
**Pull request by aeiouaeiouaeiouaeiouaeiouaeiou**
* switch to GitHub, remove platform and obsolete patch

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.8.5 12F2560 x86_64
Xcode 5.1.1 5B1008
GitHub Actions CI build

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by mascguy** [comment]
@evanmiller Can you confirm whether this port builds successfully for PPC? (Both before and after this change.)

File: audio/faad2/Portfile [hunk 1]
```diff
@@ -1,10 +1,12 @@
 # -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
 
-PortSystem      1.0
+PortSystem          1.0
+PortGroup           github 1.0
+
+github.setup        knik0 faad2 2_10_0
```
Comment (pmetzger) [inline audio/faad2/Portfile:6]: What version number is MacPorts going to get from this given the underscores? Ideally we want it to think it's 2.10.0 and not 2_10_0

**Review by pmetzger — COMMENTED** [review]
Thanks for the update. In addition to slightly inconsistent indentation, see my more substantive comment below.

File: audio/faad2/Portfile (hunk 1 above)
Comment (aeiouaeiouaeiouaeiouaeiouaeiou) [reply audio/faad2/Portfile:6]: GitHub uses [tag names](https://github.com/knik0/faad2/tags) for archives. The old repository on Sourceforge is abandoned.

**Comment by evanmiller** [comment]
> @evanmiller Can you confirm whether this port builds successfully for PPC? (Both before and after this change.)

Old version builds OK - I haven't tested the update.

**Comment by evanmiller** [comment]
> > @evanmiller Can you confirm whether this port builds successfully for PPC? (Both before and after this change.)
> 
> Old version builds OK - I haven't tested the update.

It looks like the PPC patch was merged so I'll assume the new version is fine; will open an issue later if not. The project does use the `WORDS_BIGENDIAN` macro, so there may be some issues when building PPC/Intel universal. See e.g. @ryandesign's comments in https://github.com/twogood/unshield/issues/139

File: audio/faad2/Portfile (hunk 1 above)
Comment (mascguy) [reply audio/faad2/Portfile:6]: @pmetzger Any further concerns?

**Review by mascguy — APPROVED** [review]

**Pull request merged**